shared/x509utils/types.go

```diff
@@ -27,3 +30,40 @@ type PrivateKey interface {
 type PublicKey interface {
 	Equal(x crypto.PublicKey) bool
 }
+
+// BlockToPrivateKey parses a pem Block looking for rsa, ecdsa or ed25519 Private Keys
+func BlockToPrivateKey(block *pem.Block) (PrivateKey, error) {
+	if block.Type == "PRIVATE KEY" || strings.HasSuffix(block.Type, " PRIVATE KEY") {
+		if pk, _ := x509.ParsePKCS1PrivateKey(block.Bytes); pk != nil {
+			return pk, nil
+		} else if pk, err := x509.ParsePKCS8PrivateKey(block.Bytes); err != nil {
+			return nil, err
+		} else if key, ok := pk.(PrivateKey); ok {
+			return key, nil
+		}
+	}
+	return nil, nil
```

Comment:
maybe returning not niol, nil, but nil, fmt.Error("not Private Key") or similar...

Reply:
what about something like `var ErrIgnored = errors.New("type of value out of scope, can't give an opinion")` so the caller can combine cases?

Reply:
Works for me.